consult.el

```diff
@@ -76,10 +76,10 @@
   :type 'function)
 
 (defcustom consult-mode-histories
-  '((eshell-mode . eshell-history-ring)
-    (comint-mode . comint-input-ring)
-    (term-mode   . term-input-ring))
-  "Alist of (mode . history) pairs of mode histories.
+  '((eshell-mode eshell-history-ring eshell-history-index)
+    (comint-mode comint-input-ring comint-input-ring-index)
+    (term-mode term-input-ring nil))
```

Comment:
Could you remove the nil here, such that the elements simply are either 2- or 3-lists.